fuel-vm/src/interpreter/crypto.rs

```diff
@@ -186,21 +182,10 @@ pub(crate) fn keccak256(
         Digest,
         Keccak256,
     };
-
-    let bc = checked_add_word(b, c)?;
-
-    if a > checked_sub_word(VM_MAX_RAM, Bytes32::LEN as Word)?
-        || c > MEM_MAX_ACCESS_SIZE
-        || bc > MIN_VM_MAX_RAM_USIZE_MAX
-    {
-        return Err(PanicReason::MemoryOverflow.into())
-    }
-
-    let (a, b, bc) = (a as usize, b as usize, bc as usize);
+    let src_range = MemoryRange::new(b, c)?;
```

Comment:
What happened to the check for `MEM_MAX_ACCESS_SIZE`?

Reply:
`MIN_VM_MAX_RAM_USIZE_MAX` is not a meaningful constant to have. If `VM_MAX_RAM > usize::MAX`, the VM will not work anyways with the current memory handling; it's not possible to allocate more than `usize::MAX` bytes of memory. It should be removed completely in a follow-up PR.

As for `MEM_MAX_ACCESS_SIZE`, the check was likely a mistake. The tricky part here is that it's impossible to catch this with tests, as the configuration value for it is `MEM_MAX_ACCESS_SIZE = VM_MAX_RAM`. The rule against exceeding `VM_MAX_RAM` always trigger first. Before this PR, the `MaxMemoryAccess` error was completely unused. [I asked about this in slack](https://fuellabs.slack.com/archives/C01L1HUJSL8/p1689365109352789), and after re-reading the discussion now, I feel like we're removing the it completely.

Reply:
And both removed, in 84cfbca and f0e5367.